catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/SortedQueryMonitor.java

```diff
@@ -128,10 +129,10 @@ public void run() {
           comparator = new DistanceResultComparator(sortOrder);
         } else if (Result.RELEVANCE.equals(sortType)) {
           comparator = new RelevanceResultComparator(sortOrder);
+        } else {
+          comparator = new BasicAttributeComparator(sortingProp.getPropertyName(), sortOrder);
```

Comment:
Suggestion, the above class could be simplified with:
```
comparator = ComparatorUtils.transformedComparator(
      ComparatorUtils
          .nullLowComparator(
              (sortOrder == sortOrder.ASCENDING) ? ComparatorUtils.NATURAL_COMPARATOR
                  : ComparatorUtils.reversedComparator(ComparatorUtils.NATURAL_COMPARATOR)),
      new Transformer<Result, Serializable>() {
        @Override
        public Serializable transform(Result r) {
          if (r == null) {
            return null;
          }
          final Attribute a = r.getMetacard().getAttribute(sortingProp.getPropertyName());

          return (a != null) ? a.getValue() : null;
        }
      });```

Reply:
Curious : What is the benefit to condensing the code in this way?

Reply:
Is this a ✏️  comment?

Reply:
Less code to maintain since you are re-using existing code in Java. The above already had one bug. This code simply takes advantage of the Comparator.nullsFirst() and the Comparator.comparing() methods

Reply:
There's no logic to read through , which makes it simpler to understand and maintain.

Reply:
Updating the code to match @paouelle suggestion and removing BasicAttributeComparator.java